worlds/tracker/docs/map-integration.md

```diff
@@ -0,0 +1,92 @@
+
+By defining a structure with the following fields, the world informs UT that a poptracker pack should be loaded
+
+```py
+from typing import ClassVar
+
+    tracker_world: ClassVar = {
+        "map_page_folder" : <Name of the folder that has all of poptracker files in it, not used for external poptracker packs>
+        "map_page_maps" : <Relative location(s) of the maps.json file(s), may be a list if more then one file exists>
+        "map_page_locations" : <Relative location(s) of the locations.json file(s), may be a list if more then one file exists>
+        "map_page_setting_key" : <optional tag that informs which data storage key will be watched for auto tabbing>
+        "map_page_index" : <optional function that will control the auto tabbing>
+        "external_pack_key" : <optional string that is the name of the setting string that UT reads in order to find the external pop tracker pack>
+        "poptracker_name_mapping" : <optional Dict that maps the poptracker pack names to the location id as they exist in the datapackage >
+        "location_setting_key" : <Data storage key used to determine where to place the location indicator>
+        "location_icon_coords" : <optional function used to convert between the map and the value in data storage into coords>
+
+    }
+```
+
+The setting key values have two special keys that UT will replace with the correct values in run time (because the struct needs to be a static class var)
+
+ * `{player}` : replaced with the external player slot number  
+ * `{team}` : replaced with the external team number (almost always going to be 1)  
+
+ *Note*: These are not f string values, these are literal string values on the world side
+
+for `external_pack_key` you can define the setting like this, this should point to a zipped poptracker pack
+```py
+from settings import FilePath
+class UTPackPath(FilePath):
+    #required = False #You can uncomment this to allow users to not have the poptracker map, if the key is "" then the map tab won't be rendered
+    pass
+```
+
+The contents of maps.json and locations.json are the same as poptracker format with the exception that all logic is derived from UT's internal world, and the location names must match exactly with AP location names
+
+## Implementing Auto tabbing
+
+UT can support automatically changing the loaded map tab via data storage keys, to do so you need to define the key in `map_page_setting_key` who's value will be passed to the function defined in `map_page_index`
+
+`map_page_index` is a function with the following template
+
+```py
+def map_page_index(data: Any) -> int:
+    #do code here
+    return 0
+```
+
+The function has the task to convert the value retrieved from datastorage and convert that into the index in the maps.json that should be loaded, because of the free-form nature of datastorage it is dificult to have a good general example, as the values will be dependent on what the client has access to
```

Comment:
```suggestion
The function has the task to convert the value retrieved from datastorage and convert that into the index in the maps.json that should be loaded. Because of the free-form nature of datastorage it is difficult to have a good general example, as the values will be dependent on what the client has access to.
```